xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/ActionFilter.java

```diff
@@ -88,12 +90,21 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
     {
         // Only HTTP requests can be dispatched.
         if (request instanceof HttpServletRequest
-            && !Boolean.valueOf((String) request.getAttribute(ATTRIBUTE_ACTION_DISPATCHED))) {
+            && !Boolean.parseBoolean((String) request.getAttribute(ATTRIBUTE_ACTION_DISPATCHED)))
+        {
             HttpServletRequest hrequest = (HttpServletRequest) request;
             Enumeration<String> parameterNames = hrequest.getParameterNames();
             while (parameterNames.hasMoreElements()) {
                 String parameter = parameterNames.nextElement();
-                if (parameter.startsWith(ACTION_PREFIX)) {
+                /*
+                 * If some xactions are passed as parameter, the parameters prefixed with 'action_' are only taken
+                 * into account if they are part of the xaction list.
+                 * Otherwise, all the parameters prefixed with 'action_' are accepted.
+                 */
```

Comment:
I don't know if it's documented somewhere but we normally use single line comments for inner code, and leave multi-line comments for JavaDoc.